pkg/bundles/kontext/bundle.go

```diff
@@ -44,44 +46,56 @@ func bundle(directory string) (v1.Layer, error) {
 	tw := tar.NewWriter(buf)
 	defer tw.Close()
 
-	err := filepath.Walk(directory,
-		func(path string, fi os.FileInfo, err error) error {
-			if err != nil {
-				return err
-			}
+	// build the ignore.DirectoryScanner
+	basedir, err := filepath.Abs(directory)
+	if err != nil {
+		return nil, err
+	}
+	ignorer, err := ignore.NewOrDefault(basedir)
 
-			// Skip anything in the .git directory
-			// TODO(mattmoor): expand this to .gitignore / .dockerignore?
-			if fi.IsDir() && filepath.Base(path) == ".git" {
-				return filepath.SkipDir
-			}
+	if err != nil {
+		return nil, err
+	}
 
-			// Chase symlinks.
-			info, err := os.Stat(path)
-			if err != nil {
-				return err
-			}
+	// scan the directory and build the include files, possibly filter them
+	// based on .dockerignore
+	includes, err := ignorer.Scan()
```

Comment:
I really prefer what we had before at this level 😞

Reply:
Any specific reason, I thought hiding the walk behind the scanner is a better seperation of concern. Just having only files helps to keep bundling clean

Reply:
@mattmoor  reverted the include file logic to older style using enums